### PR TITLE
L2 shorts base collateral manager

### PR DESCRIPTION
**Pull request by jacko125**
Draft PR - WIP

Refactoring CollateralManager to support L1 Inverse synths for short rates calculation and L2. 

CollateralManagerWithInverse will be able to support L1 for Inverse synths 

- BaseCollateralManager
- CollateralManagerWithInverse
- CollateralManager (L2 shorts)

### CODE REVIEW

File: publish/deployed/local-ovm/params.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 const w3utils = require('web3-utils');
 
 const params = {
-	INITIAL_ISSUANCE: '0',
+	INITIAL_ISSUANCE: w3utils.toWei('10000000'), // owner gets the SNX
```
Comment (i-stam) [inline publish/deployed/local-ovm/params.js:5]: Why did you change this? I think that with the new testing infra this is handled by either depositing to L2 or via hack-minting.

**Review by i-stam — COMMENTED** [review]
lgtm,  just one comment

File: publish/deployed/local-ovm/params.js (hunk 1 above)
Comment (jacko125) [reply publish/deployed/local-ovm/params.js:5]: @i-stam  That's right, will remove this initial issuance hack to issue owner snx as the new Infra will allow `ensureBalances()` to give the SNX to owner instead.

**Comment by jacko125** [comment]
We will be deprecating L1 inverse synths shortly. We probably won't need this PR to split the CollateralManager for L1 and L2 to be handle inverse synths.
